opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/RestTemplateAutoConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package io.opentracing.contrib.spring.web.autoconfig;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.web.client.RestTemplate;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.client.TracingRestTemplateInterceptor;
+
+/**
+ * @author Pavol Loffay
+ */
+@Configuration
+public class RestTemplateAutoConfiguration {
+
+    @Autowired(required = false)
+    private Set<RestTemplate> restTemplates;
+
+    @Autowired
+    private Tracer tracer;
+
+    @PostConstruct
+    public void init() {
+        if (restTemplates != null) {
+            for (RestTemplate restTemplate: restTemplates) {
+                registerTracingInterceptor(restTemplate);
+            }
+        }
+    }
+
+    private void registerTracingInterceptor(RestTemplate restTemplate) {
+        List<ClientHttpRequestInterceptor> interceptors = restTemplate.getInterceptors();
```

Comment:
This method is checking whether the `RestTemplate` has already been initialised with the interceptor, which indicates that the same `RestTemplate` could be initialised multiple times - so should it be synchronized to avoid concurrent updates?